cub/cub/device/dispatch/tuning/tuning_reduce.cuh

```diff
@@ -279,6 +372,94 @@ struct policy_hub
 
   using MaxPolicy = Policy1000;
 };
+
+struct arch_policies // equivalent to the policy_hub, holds policies for a bunch of CUDA architectures
```

Comment:
This is an internal type, but one that still materializes when users invoke the algorithms, right? I wonder if this should turn into a template and its data members should be turned into an environment returning those values by queries, because as is, any change to the layout would be an ABI break...

Reply:
A very appealing aspect of the current design is that tuning information is expresse very simply as structs with data members, so I would love if we could keep that.

Regarding API breaks, we do allow those at every release. This is pointed out in our [README](https://github.com/NVIDIA/cccl?tab=readme-ov-file#application-binary-interface-abi):

> Symbols in the `thrust::` and `cub::` namespaces may break ABI at any time without warning.